File: java/socketd/src/main/java/org/noear/socketd/transport/core/Handshake.java
package org.noear.socketd.transport.core;

import java.net.URI;
import java.util.Map;

/**
 * 握手信息
 *
 * @author noear
 * @since 2.0
 */
public interface Handshake {
    /**
     * 协议版本
     */
    String version();

    /**
     * 获请传输地址
     *
     * @return tcp://192.168.0.1/path?user=1&path=2
     */
    URI uri();

    /**
     * 请求路径
     */
    String path();

    /**
     * 获取参数集合
     */
    Map<String, String> paramMap();

    /**
     * 获取参数
     *
     * @param name 参数名
     */
    String param(String name);

    /**
     * 获取参数或默认值
     *
     * @param name 参数名
     * @param def  默认值
     */
    String paramOrDefault(String name, String def);

    /**
     * 放置参数
     */
    Handshake paramPut(String name, String value);

    /**
     * 输出元信息
     */
    void outMeta(String name, String value);
}

File: javascript/ts/org/noear/socketd/transport/core/Handshake.ts
/**
 * 握手信息
 *
 * @author noear
 * @since 2.0
 * @class
 */
export interface Handshake {
    /**
     * 协议版本
     * @return {string}
     */
    version(): string;

    /**
     * 获请传输地址
     *
     * @return {java.net.URI} tcp://192.168.0.1/path?user=1&path=2
     */
    uri(): URL;

    /**
     * 获取参数集合
     * @return {*}
     */
    paramMap(): any;

    /**
     * 获取参数或默认值
     *
     * @param {string} name 参数名
     * @param {string} def  默认值
     * @return {string}
     */
    paramOrDefault(name: string, def: string): string;

    /**
     * 设置或修改参数
     * @param {string} name
     * @param {string} value
     */
    param(name?: any, value?: any);
}